core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1914,7 +1974,16 @@ class Log(@volatile var dir: File,
     }
   }
 
-  override def toString = "Log(" + dir + ")"
+  override def toString: String = {
+    val logString = new StringBuilder
+    logString.append(s"Log(dir=$dir")
+    logString.append(s", topic=${topicPartition.topic}")
+    logString.append(s", partition=${topicPartition.partition}")
+    logString.append(s", highWatermark=$highWatermarkMetadata")
```

Comment:
Since we added high watermark and LSO, perhaps we may as well add the log start and end offsets as well.

Reply:
Done.